server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -181,113 +181,117 @@ void execute() {
             nodesStatsRequest.clear();
             nodesStatsRequest.addMetric(NodesStatsRequest.Metric.FS.metricName());
             nodesStatsRequest.timeout(fetchTimeout);
-            client.admin().cluster().nodesStats(nodesStatsRequest, ActionListener.runAfter(new ActionListener<>() {
-                @Override
-                public void onResponse(NodesStatsResponse nodesStatsResponse) {
-                    logger.trace("received node stats response");
+            try (var ignored = threadPool.getThreadContext().clearTraceContext()) {
+                client.admin().cluster().nodesStats(nodesStatsRequest, ActionListener.runAfter(new ActionListener<>() {
```

Comment:
This change is fairly extreme in terms of how noisy it is line wise.
Maybe, also in the sense of future readability of this code, could we extract what's inside the `try-with-resources` into a new method here and in the other spots that got indented so extremely now? That way the change-set stays small because the code doesn't move in terms of indent level and things don't get much more complicated.

Reply:
I refactored most of these sites with the extra wrapping to extract methods.